src/ode_trainer.py

```diff
@@ -0,0 +1,204 @@
+import argparse
+import os
+
+import torch
+import numpy as np
+import torch.nn as nn
+from torch.autograd import Variable
+import torch.nn.functional as F
+
+from common.settings_ode import ode_parameter_limits as ps_ode
+from common.settings_sed import p8_limits as ps_sed
+from common.settings_sed import SED_ENERGY_MIN, SED_ENERGY_MAX, SED_ENERGY_DELTA
+from common.utils import *
+from sed import sed_numba
+from models import *
+from ode import *
+
+# check for CUDA
+if torch.cuda.is_available():
+    cuda = True
+    device = torch.device("cuda")
+    torch.set_default_tensor_type(torch.cuda.FloatTensor)
+else:
+    cuda = False
+    device = torch.device("cpu")
+    torch.set_default_tensor_type(torch.FloatTensor)
+
+def generate_training_data(config):
+    train_set_size = config.train_set_size
+
+    sed_vector = []
+    haloMassLog = np.random.uniform(ps_sed[0][0], ps_sed[0][1], size=(train_set_size, 1))
+    redshift = np.random.uniform(ps_sed[1][0], ps_sed[1][1], size=(train_set_size, 1))
+    sourceAge = np.random.uniform(ps_sed[2][0], ps_sed[2][1], size=(train_set_size, 1))
+    qsoAlpha = np.random.uniform(ps_sed[3][0], ps_sed[3][1], size=(train_set_size, 1))
+    qsoEfficiency = np.random.uniform(ps_sed[4][0], ps_sed[4][1], size=(train_set_size, 1))
```

Comment:
Might be better to use Latin hyper cube sampling again as we did in the first ML-RT project (via the pyDOE package)